### ISSUE TITLE
Out of gas returns EOSIO error instead

### ISSUE THREAD

**Issue by tomtomcrypto** [telosnetwork/telos-evm-rpc#84]
When a query is out of gas we seem to return an EOSIO error instead of "Out of gas"

![photo_2024-05-20_18-47-22](https://github.com/telosnetwork/telos-evm-rpc/assets/5913758/18af627f-c4f1-43a7-90f3-a5c7f6820c43)

**Comment by tomtomcrypto** [comment]
Forcing an out of gas with

```
    // Deploy ERC20Mock contract
    const ERC20Mock = await ethers.getContractFactory("ERC20Mock");
    const erc20Mock = await ERC20Mock.deploy("MyToken", "MTK", 1000000);
    await erc20Mock.deployed();
    console.log("ERC20Mock deployed to:", erc20Mock.address);

    // Transfer tokens
    const amount = ethers.utils.parseEther("1000");
    await erc20Mock.mint("0x9a469d1e668425907548228EA525A661FF3BFa2B", amount, {
        gasLimit: 21644,
      });
    console.log("Transferred", ethers.utils.formatEther(amount), "tokens to", "0x9a469d1e668425907548228EA525A661FF3BFa2B");
```

Against master branch works:

```
ERC20Mock deployed to: 0xf2e026E0C68017F9b6f846A35620b06B3C85A171
ProviderError: VM Exception while processing transaction: Out of Gas!
    at HttpProvider.request (/home/tom/telosOFT/node_modules/hardhat/src/internal/core/providers/http.ts:90:21)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at EthersProviderWrapper.send (/home/tom/telosOFT/node_modules/hardhat-deploy-ethers/src/internal/ethers-provider-wrapper.ts:13:20)
```

**Comment by tomtomcrypto** [comment]
Forcing a wrong gasPrice doesn't tho

```
ProviderError: eosio_assert_message assertion failure at /v1/chain/send_transaction
    at HttpProvider.request (/home/tom/telosOFT/node_modules/hardhat/src/internal/core/providers/http.ts:90:21)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at EthersProviderWrapper.send (/home/tom/telosOFT/node_modules/hardhat-deploy-ethers/src/internal/ethers-provider-wrapper.ts:13:20)
```

**Comment by tomtomcrypto** [comment]
I fixed the parsing of EOSIO error, so gasPrice now gives

```
ProviderError: Transaction gas price 4 is less than the current fixed gas price of 113378400387
    at HttpProvider.request (/home/tom/telosOFT/node_modules/hardhat/src/internal/core/providers/http.ts:90:21)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at EthersProviderWrapper.send (/home/tom/telosOFT/node_modules/hardhat-deploy-ethers/src/internal/ethers-provider-wrapper.ts:13:20)
```

It will also give the proper error from EOSIO for that out of gas problem, if it wasn't something else entirely. We can fix it then at least.
